server/channels/app/platform/metrics.go

```diff
@@ -35,6 +39,8 @@ type platformMetrics struct {
 
 	cfgFn      func() *model.Config
 	listenAddr string
+
+	getPluginsEnv func() *plugin.Environment
```

Comment:
Perhaps I am missing a reason here. But why can't we simply add a method to the struct? Is it because we don't want to include platformService in platformMetrics?

Reply:
Right given `PlatformService` includes `platformMetrics` it seemed cleaner to only pass what's necessary to the latter rather than including the "parent" struct.